test/scripts/upgrade-ovn.sh

```diff
@@ -116,16 +115,167 @@ kubectl_wait_for_upgrade(){
+  NET_CIDR_IPV6=${NET_CIDR_IPV6:-fd00:10:244::/48}
+  SVC_CIDR_IPV6=${SVC_CIDR_IPV6:-fd00:10:96::/112}
+  JOIN_SUBNET_IPV4=${JOIN_SUBNET_IPV4:-100.64.0.0/16}
+  JOIN_SUBNET_IPV6=${JOIN_SUBNET_IPV6:-fd98::/64}
+  KIND_NUM_MASTER=1
+  if [ "$OVN_HA" == true ]; then
+    KIND_NUM_MASTER=3
+    KIND_NUM_WORKER=${KIND_NUM_WORKER:-0}
+  else
+    KIND_NUM_WORKER=${KIND_NUM_WORKER:-2}
+  fi
+  OVN_HOST_NETWORK_NAMESPACE=${OVN_HOST_NETWORK_NAMESPACE:-ovn-host-network}
+  OCI_BIN=${KIND_EXPERIMENTAL_PROVIDER:-docker}
+}
+
+print_ovn_manifest_params() {
+     echo "Using these parameters to build upgraded ovn-k manifests"
+     echo ""
+     echo "KIND_IPV4_SUPPORT = $KIND_IPV4_SUPPORT"
+     echo "KIND_IPV6_SUPPORT = $KIND_IPV6_SUPPORT"
+     echo "OVN_HA = $OVN_HA"
+     echo "OVN_GATEWAY_MODE = $OVN_GATEWAY_MODE"
+     echo "OVN_HYBRID_OVERLAY_ENABLE = $OVN_HYBRID_OVERLAY_ENABLE"
+     echo "OVN_DISABLE_SNAT_MULTIPLE_GWS = $OVN_DISABLE_SNAT_MULTIPLE_GWS"
+     echo "OVN_DISABLE_PKT_MTU_CHECK = $OVN_DISABLE_PKT_MTU_CHECK"
+     echo "OVN_NETFLOW_TARGETS = $OVN_NETFLOW_TARGETS"
+     echo "OVN_SFLOW_TARGETS = $OVN_SFLOW_TARGETS"
+     echo "OVN_IPFIX_TARGETS = $OVN_IPFIX_TARGETS"
+     echo "OVN_EMPTY_LB_EVENTS = $OVN_EMPTY_LB_EVENTS"
+     echo "OVN_MULTICAST_ENABLE = $OVN_MULTICAST_ENABLE"
+     echo "OVN_IMAGE = $OVN_IMAGE"
+     echo "MASTER_LOG_LEVEL = $MASTER_LOG_LEVEL"
+     echo "NODE_LOG_LEVEL = $NODE_LOG_LEVEL"
+     echo "DBCHECKER_LOG_LEVEL = $DBCHECKER_LOG_LEVEL"
+     echo "OVN_LOG_LEVEL_NORTHD = $OVN_LOG_LEVEL_NORTHD"
+     echo "OVN_LOG_LEVEL_NB = $OVN_LOG_LEVEL_NB"
+     echo "OVN_LOG_LEVEL_SB = $OVN_LOG_LEVEL_SB"
+     echo "OVN_LOG_LEVEL_CONTROLLER = $OVN_LOG_LEVEL_CONTROLLER"
+     echo "OVN_HOST_NETWORK_NAMESPACE = $OVN_HOST_NETWORK_NAMESPACE"
+     echo "OVN_ENABLE_EX_GW_NETWORK_BRIDGE = $OVN_ENABLE_EX_GW_NETWORK_BRIDGE"
+     echo "OVN_EX_GW_NETWORK_INTERFACE = $OVN_EX_GW_NETWORK_INTERFACE"
+     echo ""
+}
+
+set_cluster_cidr_ip_families() {
+  if [ "$KIND_IPV4_SUPPORT" == true ] && [ "$KIND_IPV6_SUPPORT" == false ]; then
+    IP_FAMILY=""
+    NET_CIDR=$NET_CIDR_IPV4
+    SVC_CIDR=$SVC_CIDR_IPV4
+    echo "IPv4 Only Support: API_IP=$API_IP --net-cidr=$NET_CIDR --svc-cidr=$SVC_CIDR"
+  elif [ "$KIND_IPV4_SUPPORT" == false ] && [ "$KIND_IPV6_SUPPORT" == true ]; then
+    IP_FAMILY="ipv6"
+    NET_CIDR=$NET_CIDR_IPV6
+    SVC_CIDR=$SVC_CIDR_IPV6
+    echo "IPv6 Only Support: API_IP=$API_IP --net-cidr=$NET_CIDR --svc-cidr=$SVC_CIDR"
+  elif [ "$KIND_IPV4_SUPPORT" == true ] && [ "$KIND_IPV6_SUPPORT" == true ]; then
+    IP_FAMILY="dual"
+    NET_CIDR=$NET_CIDR_IPV4,$NET_CIDR_IPV6
+    SVC_CIDR=$SVC_CIDR_IPV4,$SVC_CIDR_IPV6
+    echo "Dual Stack Support: API_IP=$API_IP --net-cidr=$NET_CIDR --svc-cidr=$SVC_CIDR"
+  else
+    echo "Invalid setup. KIND_IPV4_SUPPORT and/or KIND_IPV6_SUPPORT must be true."
+    exit 1
+  fi
+}
+
+# This script is responsible to upgrade ovn images and relevant resources from master to pr branch
```

Comment:
This is a nit.
Could this script be used locally to test upgrades manually?
In that case, using more abstracted comments that don't  allude to master and pr branch could be more helpful, like for example
` This script is responsible to upgrade ovn images and relevant resources of a running cluster to the checked-out branch`

Reply:
Fixed some of the comments here